README.md

```diff
@@ -1296,6 +1299,146 @@ CREATE VIEW animal AS
 ```
 At this point, the `animal` view can be used in the SQLAlchemy Table for the purposes of compiling.
 
+## Macro System
+Commonly used patterns in queries can be made into macros and reused. This can be used to make
+complicated queries easy to read, and make the schema more discoverable. Currently we have one
+kind of macros, macro edges.
+
+### Macro Edges
+Macro edges define a new edge in the schema by specifying how to compute it.
+Macro edges are similar to views in SQL. However, there's a few differences:
+- Macros are stored in a macro registry, that does not necessarily live inside the database. This has
+two avantages:
+  - Macros can be created locally by clients without database access.
+  - Macros work even on schemas that have been stitched from the schemas of multiple backends.
+- When a query with macros is compiled, a preprocessing step generates an equivalent query without
+macros by expanding.
+
+As an example, consider the following query:
+```graphql
+{
+    Animal {
+        name @output(out_name: "animal_name")
+        out_Animal_ParentOf {
+            out_Animal_ParentOf {
+                name @output(out_name: "grandchild_name")
+            }
+        }
+    }
+}
+```
+Let's introduce a macro for `out_Animal_GrandparentOf` and simplify our query:
+```graphql
+{
+    Animal @macro_edge_definition(name: "out_Animal_GrandparentOf") {
+        out_Animal_ParentOf {
+            out_Animal_ParentOf @macro_edge_target {
+                uuid
+            }
+        }
+    }
+}
+```
+The directive `@macro_edge_definition` declares that this is a macro, and gives it a name.
+To use it as any other edge, we will refer to it by the name given here.
+The directive `@macro_edge_target` specifies where the grandchild of the animal is.
+Now let's use this macro to rewrite our query:
+```graphql
+{
+    Animal {
+        name @output(out_name: "animal_name")
+        out_Animal_GrandparentOf {
+            name @output(out_name: "grandchild_name")
+        }
+    }
+}
+```
+Macro edges can make use of tags, filters, etc, to define more complicated macros like this one
```

Comment:
Remove the multiple filters and tags. Doesn't make the example any more educational.  I would prefer if you showed me how using macros with tags work and how tag name resolution works. Feel free to add this in your end-to-end example below.

Reply:
Why do you care about name resolution? That will just clutter the main point of macros.

Reply:
I am sorry my bad when you mentioned tags for some reason I was thinking of runtime parameters instead.